src/dali_executor/dali_pipeline.h

```diff
@@ -226,14 +228,14 @@ class DaliPipeline {
   void CreatePipeline() {
     daliCreatePipeline(&handle_, serialized_pipeline_.c_str(), serialized_pipeline_.length(),
                        max_batch_size_, num_threads_, device_id_, 0, 1, 0, 0, 0);
-    assert(handle_.pipe != nullptr && handle_.ws != nullptr);
+    dali_pipeline_created_ = true;
   }
 
 
   void ReleasePipeline() {
-    if (handle_.pipe && handle_.ws) {
-      daliDeletePipeline(&handle_);
-    }
+    if (!dali_pipeline_created_) return;
+    daliDeletePipeline(&handle_);
+    dali_pipeline_created_ = false;
```

Comment:
```suggestion
```suggestion
    if (handle_) {
      daliDeletePipeline(&handle_);
      handle_ = nullptr;
   }
```